nextcord/automod.py

```diff
@@ -0,0 +1,351 @@
+            self.keyword_filters = trigger_metadata["keyword_filter"]  # type: ignore
+        if trigger_metadata.get("presets") is not None:
+            self.presets = [try_enum(KeywordPresetType, preset) for preset in trigger_metadata["presets"]]  # type: ignore -- pylint messed up somehow
+
+    async def delete(self):
+        """|coro|
+
+        Delete this auto moderation rule."""
+        await self._state.http.delete_automod_rule(guild_id=self.guild.id, rule_id=self.id)
+
+    @property
+    def creator(self) -> Optional[Member]:
+        """Optional[:class:`Member`]: The member that created this rule."""
+        return self.guild.get_member(self.creator_id)
+
+    @property
+    def exempt_roles(self) -> List[Role]:
+        """List[:class:`Role`]: A list of role that will not be affected by this rule. `[]` if not set."""
+        return [self.guild.get_role(exempt_role_id) for exempt_role_id in self.exempt_role_ids]  # type: ignore -- they can't be None.
+
+    @property
+    def exempt_channels(self) -> List[GuildChannel]:
+        """List[:class:`GuildChannel`]: A list of channels that will not be affected by this rule. `[]` if not set."""
+        return [self.guild.get_channel(exempt_channel_id) for exempt_channel_id in self.exempt_channel_ids]  # type: ignore -- same
+
+    @overload
+    async def edit(
+        self,
+        *,
+        name: Optional[str] = ...,
+        event_type: Optional[EventType] = ...,
+        keyword_filters: Optional[List[str]] = ...,
+        notify_channel: Optional[GuildChannel] = ...,
+        timeout_seconds: Optional[int] = ...,
+        enabled: Optional[bool] = ...,
+        exempt_roles: Optional[List[Role]] = ...,
+        exempt_channels: Optional[List[GuildChannel]] = ...,
+        reason: Optional[str] = ...,
+    ) -> AutoModerationRule:
+        ...
+
+    @overload
+    async def edit(
+        self,
+        *,
+        name: Optional[str] = ...,
+        event_type: Optional[EventType] = ...,
+        presets: Optional[List[KeywordPresetType]] = ...,
+        notify_channel: Optional[GuildChannel] = ...,
+        timeout_seconds: Optional[int] = ...,
+        enabled: Optional[bool] = ...,
+        exempt_roles: Optional[List[Role]] = ...,
+        exempt_channels: Optional[List[GuildChannel]] = ...,
+        reason: Optional[str] = ...,
+    ) -> AutoModerationRule:
+        ...
+
+    async def edit(
+        self,
+        *,
+        name: Optional[str] = MISSING,
+        event_type: Optional[EventType] = MISSING,
+        keyword_filters: Optional[List[str]] = MISSING,
+        presets: Optional[List[KeywordPresetType]] = MISSING,
+        notify_channel: Optional[GuildChannel] = MISSING,
+        timeout_seconds: Optional[int] = MISSING,
+        enabled: Optional[bool] = MISSING,
+        exempt_roles: Optional[List[Role]] = MISSING,
+        exempt_channels: Optional[List[GuildChannel]] = MISSING,
+        reason: Optional[str] = MISSING,
```

Comment:
same reason as previous comments, `Optional[T]` = `T | None`, `MISSING` is not included so optional is not fit

```suggestion
        name: str = ...,
        event_type: EventType = ...,
        keyword_filters: List[str] = ...,
        notify_channel: GuildChannel = ...,
        timeout_seconds: int = ...,
        enabled: bool = ...,
        exempt_roles: List[Role] = ...,
        exempt_channels: List[GuildChannel] = ...,
        reason: str = ...,
    ) -> AutoModerationRule:
        ...

    @overload
    async def edit(
        self,
        *,
        name: str = ...,
        event_type: EventType = ...,
        presets: List[KeywordPresetType] = ...,
        notify_channel: GuildChannel = ...,
        timeout_seconds: int = ...,
        enabled: bool = ...,
        exempt_roles: List[Role] = ...,
        exempt_channels: List[GuildChannel] = ...,
        reason: str = ...,
    ) -> AutoModerationRule:
        ...

    async def edit(
        self,
        *,
        name: str = MISSING,
        event_type: EventType = MISSING,
        keyword_filters: List[str] = MISSING,
        presets: List[KeywordPresetType] = MISSING,
        notify_channel: GuildChannel = MISSING,
        timeout_seconds: int = MISSING,
        enabled: bool = MISSING,
        exempt_roles: List[Role] = MISSING,
        exempt_channels: List[GuildChannel] = MISSING,
        reason: str = MISSING,
```